downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,165 @@
+package org.reactome.release.downloadDirectory;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.biopax.validator.api.Validator;
+import org.biopax.validator.api.ValidatorUtils;
+import org.biopax.validator.api.beans.Validation;
+import org.biopax.validator.impl.IdentifierImpl;
+import org.jdom.*;
+import org.reactome.biopax.SpeciesAllPathwaysConverter;
+import org.reactome.biopax.SpeciesAllPathwaysLevel3Converter;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+
+public class Biopax {
+  private static final Logger logger = LogManager.getLogger();
+  static ApplicationContext ctx;
+  static boolean autofix = false;
+  static int maxErrors = 0;
+  static String profile = "notstrict";
+  static String outFormat = "xml";
+  static Validator validator;
+
+  static {
+   logger.info("Running BioPAX");
+    // Biopax validation requires loading of bio ontologies, which takes a few minutes
+    ctx = new ClassPathXmlApplicationContext(new String[]{
+      "META-INF/spring/appContext-validator.xml",
+      "META-INF/spring/appContext-loadTimeWeaving.xml"
+    });
+    validator = (Validator) ctx.getBean("validator");
+  }
+  
+  public static void execute(String username, String password, String host, String port, String database, String releaseNumber) throws Exception {
```

Comment:
I notice that many of the classes that do stuff in Download Directory have an `execute` method, but with signatures that are sometimes similar and sometimes different. Is there enough common functionality to create a common signature in an interface and then implement all of these different classes as implementations of a common interface? I'm not sure if this would work, just an idea...

Reply:
I was using the 'execute' as kind of a pseudo-"main" for each of the download directory steps, meant to make reading the code a bit easier. I'm not sure what you mean by the interface, so maybe we could talk about it soon

Reply:
In release-common-lib, I created a ReleaseStep as an abstract class: https://github.com/reactome/data-release-pipeline/blob/develop/release-common-lib/src/main/java/org/reactome/release/common/ReleaseStep.java

Other parts of data-release-pipeline, such as ChEBI Update, make use it by inheriting from this class: https://github.com/reactome/data-release-pipeline/blob/develop/chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdateStep.java

This way, every bit of functionality that _is a_ ReleaseStep automatically gets `getMySQLAdaptorFromProperties` and `loadTestModeFromProperties` for free, and must implement an `execute` method with a common signature. 

I'm not sure if that ReleaseStep is the correct interface for your Download Directory steps, so you could create your own from scratch, or create one that extends ReleaseStep and is used by the other parts of DownloadDirectory.